src/base/PolymeshTransactionBase.ts

```diff
@@ -243,6 +253,10 @@ export abstract class PolymeshTransactionBase<
     }
 
     try {
+      if (this.preRunValidation) {
+        await this.preRunValidation(true); // asProposal = true
```

Comment:
```suggestion
        await this.preRunValidation({asProposal: true});
```
Object params are a cleaner way to document this. Plus things can be added in with no refactor.